### PR TITLE
NXCM-3575, update MD5Crypt class

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

**Comment by mcculls** [comment]
+1

**Comment by peterlynch** [comment]
+1

File: nexus/nexus-core-plugins/nexus-ldap-plugin-parent/ldap-common/src/main/java/org/sonatype/security/ldap/dao/password/hash/MD5Crypt.java [hunk 1]
```diff
@@ -1,155 +1,210 @@
...
-import java.util.Random;
+import java.security.MessageDigest;
+import java.security.SecureRandom;
 
 /**
- * Unix MD5 encryption, used for /etc/passwd style passwords.
- * 
- * @author Kenney Westerhof
+ * Class containing static methods for encrypting passwords
+ * in FreeBSD md5 style.
```
Comment (velo) [inline nexus/nexus-core-plugins/nexus-ldap-plugin-parent/ldap-common/src/main/java/org/sonatype/security/ldap/dao/password/hash/MD5Crypt.java:23]: one more md5 way.... is that really needed?

I only ask cause we have one "official" md5 digester and probably a few apache/maven/sisu/guava like util accessible on our code base.

**Comment by velo** [comment]
+1

File: nexus/nexus-core-plugins/nexus-ldap-plugin-parent/ldap-common/src/main/java/org/sonatype/security/ldap/dao/password/hash/MD5Crypt.java (hunk 1 above)
Comment (mcculls) [inline nexus/nexus-core-plugins/nexus-ldap-plugin-parent/ldap-common/src/main/java/org/sonatype/security/ldap/dao/password/hash/MD5Crypt.java:23]: FYI, MD5crypt is not the same as MD5 (digest) http://www.usenix.org/event/usenix99/provos/provos_html/node10.html

**Comment by nabcos** [comment]
+1

**Pull request merged**